context/app/static/js/components/detailPage/DatasetRelationships/nodeTypes.tsx

```diff
@@ -139,13 +144,15 @@ function ProcessedDatasetNode({ data }: NodeProps<ProcessedDatasetNodeProps>) {
 type ComponentDatasetNodeProps = Node<DatasetNodeInfo, 'componentDataset'>;
 
 function ComponentDatasetNode({ data }: NodeProps<ComponentDatasetNodeProps>) {
+  const { datasetDetails, isLoading } = useProcessedDatasetDetails(data.uuid);
   return (
     <NodeTemplate
       rounded
       target
       icon={nodeIcons.componentDataset}
-      href={`#${data.name}-section`}
+      href={datasetDetails ? `#${datasetSectionId(datasetDetails, 'section')}` : undefined}
```

Comment:
I prefer to explicitly pass `undefined` in props when I can to be safe in case the `false` boolean value that the expression `false && "string"` gets `toString()`ed to `"false"` down the render chain. With that said, since this is repeated throughout this file, I'll make a helper function for this logic instead.